tools/ami-cleanup/src/cleanup/cleanup.go

```diff
@@ -0,0 +1,237 @@
+		imageSpace, err := ai.cleanupImageIfOld(ctx, ec2Client, devImage)
+		if err != nil {
+			return 0, 0, trace.Wrap(err, "failed to cleanup dev image %q", devImage.Name)
+		}
+
+		if imageSpace == 0 {
+			continue
+		}
+
+		totalBytesRecovered += imageSpace
+		totalImagesDeleted++
+		totalSnapshotCount += len(devImage.BlockDeviceMappings)
+	}
+
+	return totalBytesRecovered, totalImagesDeleted, nil
+}
+
+func (ai *ApplicationInstance) getEnabledRegions(ctx context.Context, client AccountAPI) ([]accounttypes.Region, error) {
+	return getAllWithPagination(
+		func(previousToken *string) (*string, []accounttypes.Region, error) {
+			results, err := client.ListRegions(ctx, &account.ListRegionsInput{
+				RegionOptStatusContains: []accounttypes.RegionOptStatus{
+					accounttypes.RegionOptStatusEnabled,
+					accounttypes.RegionOptStatusEnabledByDefault,
+				},
+				NextToken: previousToken,
+			})
+			if err != nil {
+				return nil, nil, trace.Wrap(err, "failed to request enabled regions")
+			}
+
+			return results.NextToken, results.Regions, nil
+		},
+	)
+}
+
+func (ai *ApplicationInstance) getDevImagesInRegion(ctx context.Context, client EC2API) ([]ec2types.Image, error) {
+	requestInput := &ec2.DescribeImagesInput{
+		Filters: []ec2types.Filter{
+			{
+				Name:   aws.String("name"),
+				Values: []string{"*dev*"},
+			},
+			{
+				Name:   aws.String("state"),
+				Values: []string{string(ec2types.ImageStateAvailable)},
+			},
+		},
+		IncludeDeprecated: aws.Bool(true),
+		IncludeDisabled:   aws.Bool(true),
+		Owners:            []string{"self"},
+	}
+
+	return getAllWithPagination(
+		func(previousToken *string) (*string, []ec2types.Image, error) {
+
+			requestInput.NextToken = previousToken
+			results, err := client.DescribeImages(ctx, requestInput)
+			if err != nil {
+				return nil, nil, trace.Wrap(err, "failed to request a dev images")
+			}
+
+			return results.NextToken, results.Images, nil
+		},
+	)
+}
+
+// Repeatedly calls `action` until the returned `nextToken` is null, and accumulates the results.
+// Useful for AWS API calls who's results may be paginated.
+func getAllWithPagination[T any](action func(previousToken *string) (nextToken *string, results []T, err error)) ([]T, error) {
+	var previousToken *string
+	var results []T
+	for {
+		nextToken, newResults, err := action(previousToken)
+		if err != nil {
+			return results, trace.Wrap(err, "failed to get the next set of results")
+		}
+
+		results = append(results, newResults...)
+		if nextToken == nil {
+			return results, nil
+		}
+		previousToken = nextToken
+	}
+}
+
+func (ai *ApplicationInstance) cleanupImageIfOld(ctx context.Context, client EC2API, image ec2types.Image) (int32, error) {
+	creationDate, err := iso8601.ParseString(*image.CreationDate)
+	if err != nil {
+		return 0, trace.Wrap(err, "failed to parse image %q creation timestamp %q as an ISO 8601 value", *image.Name, *image.CreationDate)
+	}
+
+	// If the image is less than a month old, don't do anything
```

Comment:
```suggestion
	// don't delete newer images
```

This comment won't go out of date if we change `maxImageKeepAge`